### PR TITLE
Dispatch User Prompts to Dialog Tool

### PR DESCRIPTION
**Pull request by gabyx**
Introduce a small tool "dialog" which dispatches all dialog prompts in `base-template.sh` to a script.
An example implementation with python3.6 and higher is given in `examples/tools/dialog`.

### CODE REVIEW

**Comment by gabyx** [comment]
Works nicely:
image

File: examples/tools/dialog/run [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
```
Comment (rycus86) [inline examples/tools/dialog/run]: Why is this needed? Can't we just have the Python script with executable flags? The `/usr/bin/env python` bit should take care of the interpreter, and there's nothing in there that needs `PYTHONPATH` to be set to anything special?

**Comment by rycus86** [comment]
I'm a bit concerned about trying to detect if we're running from a connected terminal in a cross-platform way. Can't we just check if we have a dialog tool configured, and if we do use that otherwise just try reading from the terminal as before?

File: examples/tools/dialog/run (hunk 1 above)
Comment (gabyx) [reply examples/tools/dialog/run]: This was a copy of my setup/. On windows its not so easy. On Linux for sure. Lets do it the easy way.

**Comment by gabyx** [comment]
> I'm a bit concerned about trying to detect if we're running from a connected terminal in a cross-platform way. Can't we just check if we have a dialog tool configured, and if we do use that otherwise just try reading from the terminal as before?

I think it should work, shouldn it?. The thing, is the dialog tool should really only be the case when running without a terminal attached. Because when running in the terminal you can just answer there instead of the GUI. Or should we do it like you said. and I can provide for my solution code in the `run` script to detect if we have a terminal attached. 
So far the provided solution works. At least on Windows with Git-Bash (from the official git install) etc.
Should also work in Linux.

**Comment by gabyx** [comment]
At least I tested it on MacOs as well.

**Comment by rycus86** [comment]
OK, I think this looks good to me, I'd just like to start with keeping everything as-is, except if you configure a dialog tool, in which case that is invoked instead of the `read -r`. Could you change it like that and squash please? Thanks!

**Comment by gabyx** [comment]
Ok, cool!

**Comment by gabyx** [comment]
Force pushed.

**Comment by rycus86** [comment]
This looks good, just needs a rebase/merge for the conflicts and a squash after perhaps. There's a few typos here and there, but I can take care of them after merging this PR.

**Comment by gabyx** [comment]
Maybe we could also show the stderr of the runned hook over the dialog tool, maybe, maybe in another PR :-)

**Comment by rycus86** [comment]
Sorry, had to revert this one, was breaking all the user prompts without the dialog tool. :(
I think the problem was here:
```
    else
        # Read from stdin (because its connected)
        # shellcheck disable=SC2059
        printf "$TEXT $HINT_TEXT [$SHORT_OPTIONS]: "
        read -r VARIABLE
    fi
```

This looks like we were loading user input into `$VARIABLE` instead of the actual variable name. I *think* `read -r $VARIABLE` could work though.
Would you like to reopen this and give it another try?
I think where this slipped through, is that we replaced the whole call to `show_prompt` in the test setup phase with the default answer. Instead of this, we should perhaps just change `call_script "$DIALOG_TOOL"` to return what we want it to return.

**Comment by rycus86** [comment]
@gabyx :arrow_up: in case you haven't seen

**Comment by gabyx** [comment]
Oh, I think you are right.

**Comment by gabyx** [comment]
I reopen this, :-)

**Comment by gabyx** [comment]
See #28
